app/controllers/rb_master_backlogs_controller.rb

```diff
@@ -40,8 +40,11 @@ class RbMasterBacklogsController < RbApplicationController
 
   def index
     @owner_backlogs = Backlog.owner_backlogs(@project)
-    @sprint_backlogs = Backlog.sprint_backlogs(@project)
-
+    unless params[:show_sub] and params[:show_sub].to_i == 1
+      @sprint_backlogs = Backlog.sprint_backlogs(@project)
+    else
+      @sprint_backlogs = Backlog.sprint_backlogs(@project,true)
+    end
```

Comment:
`params[:show_sub] and params[:show_sub].to_i == 1` or something similar is repeated over and over again so it would make sense to place it in a separate method ideally in a helper so that the other references can also access it. Arguably, it could also be accessed in the rb_master_backlogs_helper.rb which would render passing the parameter down time and time again unnecessary. But on the other hand it improves clarity so I can also see the benefit of doing that.

Regardless of whether the code is placed in a helper method, it can already be rewritten to 
```
with_subprojects = params[:show_sub] and params[:show_sub].to_i == 1
@sprint_backlogs = Backlog.sprint_backlogs(@project, with_subprojects)
```

With a helper method it might read
```
@sprint_backlogs = Backlog.sprint_backlogs(@project, with_subprojects?)
```
which's intention is very clear. 

That change would have two additional benefits:
* It would prevent having different implementations for behaviour that is intended to work the same but might turn out different because of the different implementations (e.g. https://github.com/finnlabs/openproject-backlogs/pull/233/files#diff-63063ba8e5d134d16ab606ca56288da2R44) 
* It allows changing the implementation at a single place. Especially, it would allow to change the implementation to not base the decision on a query parameter but on something that has been persisted in the session or in the database. I can particularly envision the later so that we could have a setting in the project or globally to control whether the subprojects are included.